### PR TITLE
fix(parser): reject single-quoted string as identifier

### LINKED ISSUE #3891: single quote string unqualified for table name
**Describe the bug**
RW:
SCR-20220714-p00

**To Reproduce**

**Expected behavior**
PG:
SCR-20220714-ozl

**Additional context**
Block regress test

### LINKED ISSUE #3980: Use identifier as `storage_parameter`
identifier = unquoted string | double-quoted string

Related to #3891, 
the root of the problem(#3891) is that a single-quoted string should be considered as a string literal instead of an identifier, which suggests we should reject a single-quoted string when parsing for an identifier.

However, as we currently anticipate an identifier in the `With` clause:
```
create table t(v1 int, v2 int) with ('appendonly' = true);
```
or
```
CREATE SOURCE IF NOT EXISTS src WITH ('kafka.topic' = 'abc', 'kafka.servers' = 'localhost:1001')
```
, therefore we require some other modifications besides rejecting single-quoted strings as identifiers.

It is doable, but after checking the syntax in PG, it is found that the `key` in the `With` clause is called [storage_parameter](https://pgpedia.info/s/storage-parameters.html#:~:text=creating%20a%20table%20with).

I suggest we follow this practice and only use an identifier, i.e. unquoted string or double-quoted string, for the keys in the `With` clause.

### PR DESCRIPTION
**Pull request by lmatz**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

- Summarize your change (**mandatory**)
  To solve both #3891 and #3980, we made two changes in code:
  1. A single-quoted string is not considered an `Identifier` during parsing.
  2. For key-value pair in the `With` clause, we parse it as `ObjectName` instead of `Identifier` because `.` will separate LHS and RHS as two `Ident`s instead of a single one while `ObjectName` consists of multiple `Ident`s.

   For test cases:
   1. We remove all the single quotes in the `With` clause when it is a `key`.

## Checklist

~- [ ] I have written necessary rustdoc comments~
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.
```
create source s with ( 'connector' = 'kafka', 'kafka.topic' = 'kafka_1_partition_topic', 'kafka.brokers' = '127.0.0.1:29092' ) row format json;
```
becomes:
```
create source s with ( connector = 'kafka', kafka.topic = 'kafka_1_partition_topic', kafka.brokers = '127.0.0.1:29092' ) row format json;
```

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* SQL commands, functions, and operators

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

TODO

## Refer to a related PR or issue link (optional)
close #3891 
close #3980

### CODE REVIEW

**Comment by neverchanje** [comment]
We will need to update the documents (the tutorials have hard-coded the version tags so it will not be broken) after this pr. I wonder what is the motivation for this change?

**Comment by lmatz** [comment]
Motivation:
1. Make it more compatible with PG.
2. The single-quoted string should be strictly rejected when an identifier is actually needed.

File: src/tests/regress/data/sql/boolean.tmp [hunk 1]
```diff
@@ -0,0 +1,263 @@
+SET RW_IMPLICIT_FLUSH TO true;
+--
+-- BOOLEAN
+--
+
```
Comment (xiangjinwu) [inline src/tests/regress/data/sql/boolean.tmp:5]: Is this added by accident or intention?

**Review by xiangjinwu — COMMENTED** [review]
rest lgtm

Do we need sth like a breaking-change tag to help surface this in next version?
cc @hengm3467

File: src/tests/regress/data/sql/boolean.tmp (hunk 1 above)
Comment (lmatz) [reply src/tests/regress/data/sql/boolean.tmp:5]: By accident😅, I will remove it.

**Comment by lmatz** [comment]
I will update the tutorial/demo if this gets merged.

File: src/sqlparser/src/ast/mod.rs [hunk 2]
```diff
@@ -161,6 +161,16 @@ impl fmt::Display for Ident {
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 pub struct ObjectName(pub Vec<Ident>);
 
+impl ObjectName {
+    pub fn real_value(&self) -> String {
```
Comment (neverchanje) [inline src/sqlparser/src/ast/mod.rs:165]: Should add some comments here.

**Review by neverchanje — COMMENTED** [review]
rest lgtm

**Pull request merged**